pandas/tests/io/test_fsspec.py

```diff
@@ -67,9 +67,38 @@ def test_reasonable_error(monkeypatch, cleared_fs):
         assert err_mgs in str(e.value)
 
 
-def test_to_csv(cleared_fs):
-    df1.to_csv("memory://test/test.csv", index=True)
-    df2 = read_csv("memory://test/test.csv", parse_dates=["dt"], index_col=0)
+@pytest.mark.parametrize(
+    "fsspec_representation,binary_mode",
+    [
+        (str, False),
+        (pytest.importorskip("fsspec").open, False),
+        (pytest.importorskip("fsspec").open, True),
+    ],
+)
+def test_to_csv(cleared_fs, fsspec_representation, binary_mode):
```

Comment:
@martindurant if you'd have a quick look here

Reply:
Yes it's fine, but a little hard to read. May be clearer to split out the string and file-like cases into separate tests.